cpp/src/binaryop/compiled/binary_ops.cu

```diff
@@ -0,0 +1,260 @@
+  CUDA_DEVICE_CALLABLE Out greater(Lhs const& x, Rhs const& y) const { return static_cast<Out>(x > y); }
+  CUDA_DEVICE_CALLABLE Out less_equal(Lhs const& x, Rhs const& y) const { return static_cast<Out>(x <= y); }
+  CUDA_DEVICE_CALLABLE Out greater_equal(Lhs const& x, Rhs const& y) const { return static_cast<Out>(x >= y); }
+};
+
+/**
+ * @brief Computes output valid mask for op between a column and a scalar
+ */
+auto scalar_col_valid_mask_and(column_view const& col,
+                               scalar const& s,
+                               cudaStream_t stream,
+                               rmm::mr::device_memory_resource* mr) {
+  if (col.size() == 0) {
+    return rmm::device_buffer{};
+  }
+
+  if (not s.is_valid()) {
+    return create_null_mask(col.size(), mask_state::ALL_NULL, stream, mr);
+  } else if (s.is_valid() && col.nullable()) {
+    return copy_bitmask(col, stream, mr);
+  } else if (s.is_valid() && not col.nullable()) {
+    return rmm::device_buffer{};
+  }
+  return rmm::device_buffer{};
+}
+
+template <typename Lhs, typename Rhs, typename Out>
+struct binary_op {
+
+  std::unique_ptr<column> operator()(scalar const& lhs, column_view const& rhs, binary_operator op, data_type out_type, rmm::mr::device_memory_resource* mr, cudaStream_t stream) {
+    auto new_mask = scalar_col_valid_mask_and(rhs, lhs, stream, mr);
+    auto out = make_fixed_width_column(out_type, rhs.size(), new_mask,
+                                       cudf::UNKNOWN_NULL_COUNT, stream, mr);
+
+    if (rhs.size() > 0) {
+      auto out_view = out->mutable_view();
+      auto out_itr = out_view.begin<Out>();
+      auto rhs_device_view = column_device_view::create(rhs, stream);
+      auto lhs_scalar = static_cast<cudf::experimental::scalar_type_t<Lhs> const&>(lhs);
+      auto binop = apply_binop<Lhs, Rhs, Out>{op};
+      auto lhs_scalar_view = get_scalar_device_view(lhs_scalar);
+      auto apply_binop_scalar = [binop, lhs_scalar_view] __device__ (Rhs const& rhs) {
+        return binop(lhs_scalar_view.is_valid() ? lhs_scalar_view.value() : Lhs{}, rhs);
+      };
+      if (rhs.has_nulls()) {
+        auto rhs_itr = detail::make_null_replacement_iterator(*rhs_device_view, Rhs{});
+        thrust::transform(rmm::exec_policy(stream)->on(stream), rhs_itr, rhs_itr + rhs.size(), out_itr, apply_binop_scalar);
+      } else {
+        auto rhs_itr = thrust::make_transform_iterator(thrust::make_counting_iterator(size_type{0}),
+                                                      [col=*rhs_device_view] __device__ (size_type i) { return col.element<Rhs>(i); });
+        thrust::transform(rmm::exec_policy(stream)->on(stream), rhs_itr, rhs_itr + rhs.size(), out_itr, apply_binop_scalar);
+      }
+    }
+
+    CHECK_CUDA(stream);
+
+    return out;
+  }
+
+  std::unique_ptr<column> operator()(column_view const& lhs, scalar const& rhs, binary_operator op, data_type out_type, rmm::mr::device_memory_resource* mr, cudaStream_t stream) {
```

Comment:
Can this and the other scalar-column method have a common implementation along with some operator inversion logic.

Reply:
Done in https://github.com/rapidsai/cudf/pull/3657/commits/1cad797e51cf1a0f9f585cd2bda4087fcadf7f8e